rosidl_generator_cpp/resource/msg.hpp.template

```diff
@@ -15,6 +15,7 @@
 #ifndef __@(spec.base_type.pkg_name)__@(subfolder)__@(get_header_filename_from_msg_name(spec.base_type.type))__hpp__
 #define __@(spec.base_type.pkg_name)__@(subfolder)__@(get_header_filename_from_msg_name(spec.base_type.type))__hpp__
 
+#include "@(spec.base_type.pkg_name)/@(subfolder)/@(get_header_filename_from_msg_name(spec.base_type.type))__meta.hpp"
```

Comment:
I'm actually working on an example of how this trait will be used in `rclcpp` right now:

`MessageMemoryStrategy<MessageT>` is injected into `Subscription<MessageT>`
`BoundedMsgMemoryStrategy<MessageT>` inherits `MsgMemoryStrategy`
`BoundedMsgMemoryStrategy<MessageT>` has some `enable_if` logic that prevents compilation if the user tries to use the template specialization where `MessageT` is unbounded.

Like William said, the user has to remember to include the `traits` header for the message when using `BoundedMessageMemoryStrategy`.

this might be OK since message type traits are "advanced" user topics and not needed in most use cases.

But I'm inclined to agree with William, since the connection between the `traits` header and the object where it is will be used is not obvious and the potential source of many headaches.